provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -330,57 +332,64 @@ def _get_data_devices(self, machine_id: str, cvg: int) -> DList[Text]:
     # TBD motr
     def _get_metadata_device(self,
                              machine_id: str,
-                             cvg: int,
-                             m0d: int) -> Text:
-        motr_store = self.motr_provider
-        metadata_device = Text(motr_store.get(
-            f'server>{machine_id}>cvg[{cvg}]>m0d[{m0d}]>md_seg1'))
+                             cvg: int) -> Text:
+        store = self.provider
+        metadata_device = Text(store.get(
+            f'node>{machine_id}>storage>cvg[{cvg}]>devices>metadata'))
         return metadata_device
 
-    # TBD motr
+    # This function is kept as place holder with length returning 1,
+    # as policy needs to be decided for a commong solution that is
+    # applicable for LR and LC. This function can be used or removed
+    # in that task (EOS-26849)
     def _get_m0d_per_cvg(self, machine_id: str, cvg: int) -> int:
-        motr_store = self.motr_provider
-        return len(motr_store.get(f'server>{machine_id}>cvg[{cvg}]>m0d'))
+        length = 1
+        return length
 
     def _create_node(self, machine_id: str) -> NodeDesc:
         store = self.provider
 
         hostname = self.utils.get_hostname(machine_id)
         # node>{machine-id}>name
         iface = self._get_iface(machine_id)
-        try:
-            # cortx>motr>client_instances
-            no_m0clients = int(store.get(
-                'cortx>motr>client_instances',
-                allow_null=True))
-        except TypeError:
-            no_m0clients = 2
-        # Currently, there is 1 m0d per cvg.
-        # We will create 1 IO service entry in CDF per cvg.
-        # An IO service entry will use data devices from corresponding cvg.
-        # meta data device is taken from motr-hare shared store.
-        servers = DList([
-            M0ServerDesc(
+        servers = None
+        no_m0clients = 0
```

Comment:
We are hard coding this, same for the s3 instances, We need to request these keys, cortx>motr>client_instances and cortx>s3>service_instances to move under the node hierarchy.
Or we need to check if a node is running S3 services then use cortx>s3>service_instances and create those many number of `m0_clients` instances similar to `m0_servers`
e.g.
```
    m0_clients:
      s3: 0                # number of S3 servers to start
      other: 2            # max number of other Motr clients this host may have
  - hostname: client-21    # [user@]hostname
    data_iface: ib1        # name of data network interface
    data_iface_type: o2ib  # type of network interface (optional);
    m0_clients:
      s3: 0                # number of S3 servers to start
      other: 16            # max number of other Motr clients this host may have
  - hostname: client-22    # [user@]hostname
    data_iface: ib1        # name of data network interface
    data_iface_type: o2ib  # type of network interface (optional);
    m0_clients:
      s3: 0                # number of S3 servers to start
      other: 16
```
These are some pure client nodes.

Reply:
We are already reading the confstore keys and returning values for "no_m0clients" and "nr_s3_instances".
"if(self.utils.is_motr_component(machine_id))" This check is reading confstore key to decide whether motr component is present or not.

Only the reason I had to put variable inits to some values here was, to have non-none values as the returning class "NodeDesc" has fixed set of parameters.

Reply:
Now this patch is updated with latest commit where we are checking s3 component separately, to fix deployement in regular way - issue.
Thanks!